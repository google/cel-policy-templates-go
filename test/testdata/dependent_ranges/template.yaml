# Copyright 2020 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#    https://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

apiVersion: policy.acme.co/v1
kind: PolicyTemplate
metadata:
  name: dependent_ranges
evaluator:
  description: >
    Ranges may depend on prior ranges as input arguments within the 'in'
    expression. In the example below, the values in 'i' are multiplied
    to produce the values for j. The result should be:
      [2, 3, 6], [3, 2, 6]

  ranges:
    - {value: i, in: "[1, 2, 3]"}
    - {index: idx, value: j, in: "[i, i * 2, i * 3]"}
  productions:
    - match: j % 6 == 0
      decision: policy.report
      output: >
        [i, idx + 1, j]
