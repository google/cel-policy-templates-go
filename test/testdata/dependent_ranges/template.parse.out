1~# Copyright 2020 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#    https://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

2~apiVersion: 3~"policy.acme.co/v1"
4~kind: 5~"PolicyTemplate"
6~metadata:7~
  8~name: 9~"dependent_ranges"
10~evaluator:11~
  12~description: 13~>
    Ranges may depend on prior ranges as input arguments within the 'in'
    expression. In the example below, the values in 'i' are multiplied
    to produce the values for j. The result should be:
      [2, 3, 6], [3, 2, 6]
  14~ranges:15~
    - 16~{17~value: 18~"i", 19~in: 20~"[1, 2, 3]"}
    - 21~{22~index: 23~"idx", 24~value: 25~"j", 26~in: 27~"[i, i * 2, i * 3]"}
  28~productions:29~
    - 30~31~match: 32~"j % 6 == 0"
      33~decision: 34~"policy.report"
      35~output: 36~>
        [i, idx + 1, j]
